src/librustc_mir/dataflow/move_paths/builder.rs

```diff
@@ -103,6 +103,8 @@ impl<'b, 'a, 'tcx> Gatherer<'b, 'a, 'tcx> {
             }
         };
 
+        let mut union_path = None;
```

Comment:
I feel like it would be good to have a comment here explaining the role of this variable -- and in particular what happens once it gets set to `Some` (e.g., `base` no longer gets updated)